pyrene/src/components/SearchFinder/SearchFinder.tsx

```diff
@@ -1,23 +1,50 @@
-import React, { useCallback } from 'react';
+import React, {
+  useCallback, FunctionComponent, KeyboardEvent, ChangeEvent,
+} from 'react';
 import clsx from 'clsx';
-import PropTypes from 'prop-types';
-
 import styles from './searchFinder.css';
 import Icon from '../Icon/Icon';
 import SearchInput from './components/SearchInput/SearchInput';
 
+export interface SearchFinderProps {
+  /**
+   * called when searchTerm changes
+   */
+  onSearchTermChange: (value: string, e?: ChangeEvent<HTMLInputElement>) => void,
+  /**
+   * called when selectedResult changes
+   */
+  onSelectedResultChange: (selectedItem: number) => void,
+  /**
+   * input placeholder string
+   */
+  placeholder?: string,
+  /**
+   * total number of results
+   */
+  resultCount: number,
+  /**
+   * displayed in search input
+   */
+  searchTerm: string,
+  /**
+   * currently selected result, must be smaller than resultCount.
+   */
+  selectedResult: number,
+}
+
 /**
  * Search input area with buttons that cycle between matches.
  * Similar to ctrl+f exact search in browsers, but searching logic is custom.
  */
-const SearchFinder = ({
+const SearchFinder: FunctionComponent<SearchFinderProps> = ({
   onSearchTermChange,
   searchTerm,
   resultCount,
   selectedResult,
-  placeholder,
+  placeholder = '',
   onSelectedResultChange,
-}) => {
+}: SearchFinderProps) => {
```

Comment:
Do you want to remove `SearchFinderProps` here like you did in other PRs?

Reply:
@cyrilwanner Here I need to keep it. Because it s an **exposed** component (not internal). Hence the `babel plugin` has to generate the `PropType` for the documentation system (`kitchensink`).

Reply:
Ah, so we can only remove it on non-exposed components for now. Cool, thank you for clarifying!